### PR TITLE
Remove akka-agent

### PR DESCRIPTION
**Pull request by raboof**
Refs #26184

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-docs/src/main/paradox/agents.md [hunk 1]
```diff
@@ -1,168 +1,5 @@
...
-transaction is aborted. @scala[Here's an example:]
-
-@@@ div { .group-scala }
-
-@@snip [AgentDocSpec.scala](/akka-docs/src/test/scala/docs/agent/AgentDocSpec.scala) { #transfer-example }
-
-@@@
+The akka-agent module was deprecated in Akka 2.5 and has been removed in
+2.6. If there is interest it may be moved to a separate, community-maintained
+repository.
```
Comment (patriknw) [inline akka-docs/src/main/paradox/agents.md]: I think we should remove the page, it's enough that this is mentioned in migration guide.
Maybe add redirect in .htaccess

File: build.sbt [hunk 2]
```diff
@@ -31,7 +31,6 @@ def isScala211: Boolean = System.getProperty("akka.build.scalaVersion", "").star
 // When this is updated the set of modules in ActorSystem.allModules should also be updated
 lazy val aggregatedProjects: Seq[ProjectReference] = List[ProjectReference](
   actor, actorTests,
-  agent,
```
Comment (patriknw) [inline build.sbt]: also remove from `allModules` in ActorSystem.scala

**Review by patriknw — COMMENTED** [review]
missing mention in migration guide

**Review by helena — APPROVED** [review]
LGTM

**Pull request merged**